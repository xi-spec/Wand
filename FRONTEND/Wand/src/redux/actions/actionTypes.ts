export default {
  LOAD_PRODUCT_TYPE: 'LOAD_PRODUCT_TYPE',
  LOAD_ALL_PRODUCTS: 'LOAD_ALL_PRODUCTS',
  LOAD_NEW_PRODUCTS: 'LOAD_NEW_PRODUCTS',
  LOAD_PRODUCTS_LIST: 'LOAD_PRODUCTS_LIST',
  LOAD_PRODUCT_DETAIL: 'LOAD_PRODUCT_DETAIL',
  ADD_PRODUCT_TO_CART: ' ADD_PRODUCT_TO_CART',
  INCREASE_QUANTITY_CART: 'INCREASE_QUANTITY_CART',
  DECREASE_QUANTITY_CART: 'DECREASE_QUANTITY_CART',
  DELETE_PRODUCT_CART: 'DELETE_PRODUCT_CART',
  DELETE_ALL_CART: 'DELETE_ALL_CART',
  UPDATE_STOCK_API: 'UPDATE_STOCK_API',
  SEARCH_PRODUCT: 'SEARCH_PRODUCT',
  ADD_PRODUCT_TO_FAVOURITE: 'ADD_PRODUCT_TO_FAVOURITE',
  REGISTER: 'REGISTER',
  REGISTER_ERROR: 'REGISTER_ERROR',
  LOGIN: 'LOGIN',
  LOGIN_ERROR: 'LOGIN_ERROR',
  CLEAN_ISEXIST_STATUS: 'CLEAN_STATUS',
  CLEAN_LOGIN_ERROR: 'CLEAN_LOGIN_ERROR',
  CLEAN_REGISTED: 'CLEAN_REGISTED',
  REMOVE_FAVOURITE: 'REMOVE_FAVOURITE',
  LOGOUT: 'LOGOUT',
  UPDATE_USER: 'UPDATE_USER',
  PAYMENT_SUCCSSFULY: 'PAYMENT_SUCCESSFULY',
  PDATE_STOCK_API: 'PDATE_STOCK_API'
}
;
