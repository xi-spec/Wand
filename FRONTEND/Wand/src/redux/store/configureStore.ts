import reduxImmutableStatInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import rootReducer from '../reducers';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(reduxImmutableStatInvariant(), thunk))
  );
}

export default configureStore();
