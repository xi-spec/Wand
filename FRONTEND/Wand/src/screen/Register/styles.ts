import { StyleSheet } from 'react-native';

import {
  COLORS,
  SIZE
} from '../../constants/theme';

const styles = StyleSheet.create({
  container: {
    marginLeft: 50,
    marginRight: 50,
    paddingTop: 50
  },
  rowContainer: {
    width: '100%',
    height: '70%'
  },
  blueBigText: {
    color: COLORS.primary,
    fontSize: 36,
    marginBottom: 5

  },
  greySmallText: {
    color: COLORS.darkGrey,
    fontSize: SIZE.primary

  },
  inputContainer: {
    paddingTop: 50
  },
  input: {
    width: 300,
    height: 50,
    fontSize: SIZE.primary,
    color: COLORS.primary,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.darkGrey,
    marginBottom: 10,
    marginTop: 10

  },
  errorContainer: {
    height: 15
  },
  wrongMessage: {
    fontSize: 12, color: COLORS.second
  },

  buttonContainer: {
    alignItems: 'center',
    marginTop: 30

  },
  button: {
    width: '100%',
    height: 50,
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: SIZE.primary,
    backgroundColor: COLORS.primary

  },

  buttonText: {
    color: 'white',
    fontSize: SIZE.buttonSize
  }

});

export default styles;
