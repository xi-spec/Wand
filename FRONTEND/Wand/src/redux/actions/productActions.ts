import actionTypes from './actionTypes';
import axios from 'axios';
import SERVER from '../../constants/server';
import { Dispatch } from 'redux';
import { Product, ShoppingCart } from '../../models/initiaState';

export function loadProductsType () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_TYPE);
    dispatch({
      type: actionTypes.LOAD_PRODUCT_TYPE,
      data
    });
  };
}

export function loadAllProducts () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_PRODUCT);
    dispatch({
      type: actionTypes.LOAD_ALL_PRODUCTS,
      data
    });
  };
}

export function loadProductsList (typeName:string) {
  return {
    type: actionTypes.LOAD_PRODUCTS_LIST,
    typeName
  };
}

export function loadProductDetail (productId:string) {
  return {
    type: actionTypes.LOAD_PRODUCT_DETAIL,
    productId
  };
}

export function addProductToCart (product:Product) {
  return {
    type: actionTypes.ADD_PRODUCT_TO_CART,
    product
  };
}

export function increaseQuantityCart (productId:string) {
  return {
    type: actionTypes.INCREASE_QUANTITY_CART,
    productId
  };
}

export function decreaseQuantityCart (productId:string) {
  return {
    type: actionTypes.DECREASE_QUANTITY_CART,
    productId
  };
}

export function deleteProductCart (productId:string, quantity:number) {
  return {
    type: actionTypes.DELETE_PRODUCT_CART,
    productId,
    quantity
  };
}

export function searchProducts (name:string) {
  return {
    type: actionTypes.SEARCH_PRODUCT,
    name
  };
}

export function updateStockApi (cart:ShoppingCart) {
  return (dispatch : Dispatch) => {
    cart.cartList.forEach(async (product) => {
      const { data } = await axios.put(`${SERVER.URL_PRODUCT}/${product._id}`, { stock: product.stock });
      dispatch({
        type: actionTypes.UPDATE_STOCK_API,
        data
      });
    });
  };
}
