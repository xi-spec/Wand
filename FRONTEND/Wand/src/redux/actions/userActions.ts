import axios from 'axios';
import actionTypes from './actionTypes';
import SERVER from '../../constants/server';
import { User, Product, ShoppingCart } from '../../models/initiaState';
import { Dispatch } from 'redux';

export function addProductToFavourite (product:Product, user:User, favouriteState:boolean) {
  let newUser;
  favouriteState
    ? newUser = { ...user, favourite: [...user.favourite, product] }
    : newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.ADD_PRODUCT_TO_FAVOURITE,
      data
    });
  };
}

export function register (email:string, name:string, password:string) {
  const query = { email, name, password };
  let actionObj;
  return async (dispatch:Dispatch) => {
    try {
      const { data, status } = await axios.post(SERVER.URL_REGISTER, query);
      actionObj = {
        type: actionTypes.REGISTER,
        data,
        status
      };
    } catch (error) {
      actionObj = {
        type: actionTypes.REGISTER_ERROR
      };
    }
    dispatch(
      actionObj
    );
  };
}

export function login (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    let actionObj;
    try {
      const { data } = await axios.post(SERVER.URL_LOGIN, query);
      actionObj = {
        type: actionTypes.LOGIN,
        data

      };
    } catch (error) {
      actionObj = {
        type: actionTypes.LOGIN_ERROR
      };
    }

    dispatch(actionObj);
  };
}

export function logout (email:string, password:string) {
  const query = { email, password };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.post(SERVER.URL_LOGOUT, query);

    dispatch({
      type: actionTypes.LOGOUT,
      data
    });
  };
}

export function cleanIsExistStatus () {
  return {
    type: actionTypes.CLEAN_ISEXIST_STATUS
  };
}

export function cleanLoginError () {
  return {
    type: actionTypes.CLEAN_LOGIN_ERROR
  };
}

export function cleanRegisted () {
  return {
    type: actionTypes.CLEAN_REGISTED
  };
}

export function removeFavourite (product:Product, user:User) {
  const newUser = { ...user, favourite: user.favourite.filter(elem => elem._id !== product._id) };

  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, newUser);
    dispatch({
      type: actionTypes.REMOVE_FAVOURITE,
      data
    });
  };
}

export function updateAddress (street:string,
  city:string, country:string, postcode:string, user:User) {
  const query = { street, city, country, postcode };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { address: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function updateCard (cardNumber:string, securtyCode:string, cardholder:string, expiryDate:string, user:User) {
  const query = { cardNumber, securtyCode, cardholder, expiryDate };
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { card: query });
    dispatch({
      type: actionTypes.UPDATE_USER,
      data
    });
  };
}

export function paymentSuccessfuly (user:User, cart:ShoppingCart) {
  return async (dispatch:Dispatch) => {
    const { data } = await axios.put(`${SERVER.URL_USER}/${user._id}`, { orderHistory: [...user.orderHistory, cart] });
    dispatch({
      type: actionTypes.PAYMENT_SUCCSSFULY,
      data
    });
  };
}
