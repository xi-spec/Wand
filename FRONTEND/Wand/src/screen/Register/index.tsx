import React from 'react';
import {
  Text,
  TouchableOpacity,
  View,
  TextInput,
  Alert
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import { COLORS } from '../../constants/theme';
import styles from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  register,
  cleanIsExistStatus,
  cleanRegisted
} from '../../redux/actions/userActions';
import {
  hasNameError,
  hasEmailError,
  hasPasswordError,
  hasConfirmPasswordError
} from '../../utils/validation';
import { User } from '../../models/initiaState';
import { ScrollView } from 'react-native-gesture-handler';

function Register ({ user, actions, navigation }:{
  user:User,
  actions:any,
  navigation:{navigate:Function}
}) {
  const [name, setName] = React.useState('');
  const [email, setEmail] = React.useState('');
  const [password, setPassword] = React.useState('');
  const [confirmPassword, setConfirmPassword] = React.useState('');
  const [nameError, setError] = React.useState(false);
  const [emailError, setEmailError] = React.useState(false);
  const [pdwError, setPdwError] = React.useState(false);
  const [confirmPwdError, setConfirmPdwError] = React.useState(false);

  function isDisabled () {
    return !name ||
      !password ||
      !email ||
      !password ||
      nameError ||
      pdwError ||
      confirmPwdError ||
      emailError;
  };

  return (
    <>
    { user.isExist && Alert.alert('', 'This email has been already used for register, please try again',
      [
        { text: 'OK', onPress: () => actions.cleanIsExistStatus() }
      ])}
      {user.isRegisted && Alert.alert('', 'Registration success! ',
        [
          { text: 'OK', onPress: () => { navigation.navigate('Login'); actions.cleanRegisted(); } }
        ]) }
  <ScrollView
  keyboardShouldPersistTaps='handled'
   style={styles.container} >

    <View >

    <Text style={styles.blueBigText}>
    Hi!
   </Text>

   <Text style={styles.greySmallText}>
  Create a new account
   </Text>
    </View>

    <View style={styles.rowContainer}>

    <TextInput
         placeholderTextColor={COLORS.darkGrey}
         placeholder="Name"
         style={styles.input}
         keyboardType="email-address"
         value={name}
         onChangeText={(text) => setName(text)}
         onEndEditing={() => setError(() => hasNameError(name))}
       >
       </TextInput>
       <View style={styles.errorContainer}>
       {nameError && <Animatable.View animation='fadeInLeft' duration={500}>
       <Text style={styles.wrongMessage}>Name must be at least 2 characters long</Text>
       </Animatable.View>}
       </View>

       <TextInput
         autoCapitalize='none'
         value={email}
         placeholderTextColor={COLORS.darkGrey}
         placeholder="Email Address"
         style={styles.input}
         keyboardType="email-address"
         onChangeText={(text) => setEmail(text)}
         onEndEditing ={() => setEmailError(hasEmailError(email))}
       />
       <View style={styles.errorContainer}>
         {emailError && <Animatable.View animation='fadeInLeft' duration={500}>
       <Text style={styles.wrongMessage}>Email is not valid</Text>
       </Animatable.View>}
         </View>

       <TextInput
         placeholderTextColor={COLORS.darkGrey}
         placeholder="Password"
         style={styles.input}
         secureTextEntry
         value={password}
         onChangeText={(text) => setPassword(text)}
         onEndEditing ={() => setPdwError(() => hasPasswordError(password))}
       />

       <View style={styles.errorContainer}>

       {pdwError && <Animatable.View animation='fadeInLeft' duration={500}>
         <Text style={styles.wrongMessage}>Password must be at least 8 characters long</Text>
         </Animatable.View> }
       </View>

        <TextInput
         placeholderTextColor={COLORS.darkGrey}
         placeholder="Confirm password"
         style={styles.input}
         secureTextEntry
         value={confirmPassword}
         onChangeText={(text) => setConfirmPassword(text)}
         onEndEditing ={() => setConfirmPdwError(() => hasConfirmPasswordError(password, confirmPassword))}
       />
       <View style={styles.errorContainer}>

       {confirmPwdError && <Animatable.View animation='fadeInLeft' duration={500}>
         <Text style={styles.wrongMessage}>Both passwords must be the same </Text>
         </Animatable.View>}
       </View>
       <View style={styles.buttonContainer}>
       <TouchableOpacity
        disabled={isDisabled()}
       style={styles.button}
       testID='register'
       onPress={() => actions.register(email, name, password)}>
           <Text style={styles.buttonText}>REGISTER</Text>
           </ TouchableOpacity>
       </View>
     </View>

</ScrollView>
  </>
  );
}
function mapStateToProps ({ user }) {
  return {
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      register,
      cleanIsExistStatus,
      cleanRegisted
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Register);
