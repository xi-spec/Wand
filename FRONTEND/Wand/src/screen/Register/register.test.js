import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import configureStore from 'redux-mock-store';
import * as actions from '../../redux/actions/userActions';
import { Provider } from 'react-redux';
import Register from './index';
import { Alert } from 'react-native';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
describe('Given a Register component', () => {
  const mockStore = configureStore();
  let store;
  let navigation;
  let alert;

  beforeEach(() => {
    jest.spyOn(actions, 'cleanIsExistStatus').mockReturnValueOnce({ type: '' });
    jest.spyOn(actions, 'register').mockReturnValueOnce({ type: '' });
    jest.spyOn(actions, 'cleanRegisted').mockReturnValueOnce({ type: '' });
    alert = jest.spyOn(Alert, 'alert');

    store = mockStore({
      user: { isLogged: false }
    });
    navigation = {
      navigate: jest.fn()
    };
  });
  describe('When is render', () => {
    describe('And user.isLogged is false', () => {
      test('Then is mathced to snapshot', () => {
        const tree = render(<Provider store={store}><Register navigation={navigation} /></Provider>);

        expect(tree).toMatchSnapshot();
      });
    });

    describe('And user.isRegisted is true', () => {
      test('Then alert is invoked', () => {
        store = mockStore({
          user: {
            isRegisted: true
          }
        });

        render(<Provider store={store}><Register navigation={navigation} /></Provider>);

        expect(alert).toHaveBeenCalled();
      });

      test('Then navigation.navigate is invoked', () => {
        render(<Provider store={store}><Register navigation={navigation} /></Provider>);
        alert.mock.calls[0][2][0].onPress();

        expect(actions.cleanRegisted).toHaveBeenCalled();
      });
    });

    describe('And user.isExist is true', () => {
      test('Then Alert.alert is invoked', () => {
        store = mockStore({
          user: {
            isExist: true
          }
        });

        render(<Provider store={store}><Register /></Provider>);

        expect(alert).toHaveBeenCalled();
      });
    });

    describe('And pressed ok button of Alert.alert', () => {
      test('Then actions.cleanIsExistStatus is invoked', () => {
        store = mockStore({
          user: {
            isExist: true
          }
        });
        alert.mock.calls[1][2][0].onPress();

        render(<Provider store={store}><Register /></Provider>);

        expect(actions.cleanIsExistStatus).toHaveBeenCalled();
      });
    });
  });

  describe('When nameTextInput is typing wiht NAME', () => {
    test('Then nameTextInput.value is NAME ', () => {
      const { getByPlaceholderText } = render(<Provider store={store}><Register /></Provider>);

      const text = 'NAME';
      const nameTextInput = getByPlaceholderText('Name');

      fireEvent.changeText(nameTextInput, text);

      expect(nameTextInput.props.value).toBe(text);
    });
  });

  describe('When some textInput is typing', () => {
    test('Then her onChange function will invoked', () => {
      const tree = render(<Provider store={store}><Register /></Provider>);

      const text = 'HELLO';

      const nameTextInput = tree.getByPlaceholderText('Name');
      fireEvent.changeText(nameTextInput, text);

      const passwordInput = tree.getByPlaceholderText('Password');
      fireEvent.changeText(passwordInput, text);

      const emailInput = tree.getByPlaceholderText('Email Address');
      fireEvent.changeText(emailInput, text);

      const confirmInput = tree.getByPlaceholderText('Confirm password');
      fireEvent.changeText(confirmInput, text);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When have textInput Error', () => {
    test('Then will render a wrong message', () => {
      const mockState = React.useState;
      jest.spyOn(React, 'useState')
        .mockImplementationOnce(() => mockState(''))
        .mockImplementationOnce(() => mockState(''))
        .mockImplementationOnce(() => mockState(''))
        .mockImplementationOnce(() => mockState(''))
        .mockImplementationOnce(() => mockState(true))
        .mockImplementationOnce(() => mockState(true))
        .mockImplementationOnce(() => mockState(true))
        .mockImplementationOnce(() => mockState(true));

      const tree = render(<Provider store={store}><Register /></Provider>);
      expect(tree).toMatchSnapshot();
    });
  });
});
