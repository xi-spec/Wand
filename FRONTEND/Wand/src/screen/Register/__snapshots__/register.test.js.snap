// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register component When have textInput Error Then will render a wrong message 1`] = `
<RCTScrollView
  collapsable={false}
  keyboardShouldPersistTaps="handled"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "marginLeft": 50,
      "marginRight": 50,
      "paddingTop": 50,
    }
  }
>
  <View>
    <View>
      <Text
        style={
          Object {
            "color": "#2E3192",
            "fontSize": 36,
            "marginBottom": 5,
          }
        }
      >
        Hi!
      </Text>
      <Text
        style={
          Object {
            "color": "#979797",
            "fontSize": 18,
          }
        }
      >
        Create a new account
      </Text>
    </View>
    <View
      style={
        Object {
          "height": "70%",
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Name"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      >
        <View
          forwardedRef={[Function]}
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F3603F",
                "fontSize": 12,
              }
            }
          >
            Name must be at least 2 characters long
          </Text>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Email Address"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      >
        <View
          forwardedRef={[Function]}
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F3603F",
                "fontSize": 12,
              }
            }
          >
            Email is not valid
          </Text>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Password"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      >
        <View
          forwardedRef={[Function]}
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F3603F",
                "fontSize": 12,
              }
            }
          >
            Password must be at least 8 characters long
          </Text>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Confirm password"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      >
        <View
          forwardedRef={[Function]}
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F3603F",
                "fontSize": 12,
              }
            }
          >
            Both passwords must be the same 
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 30,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2E3192",
              "borderRadius": 5,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 18,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="register"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            REGISTER
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Given a Register component When is render And user.isLogged is false Then is mathced to snapshot 1`] = `
<RCTScrollView
  collapsable={false}
  keyboardShouldPersistTaps="handled"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "marginLeft": 50,
      "marginRight": 50,
      "paddingTop": 50,
    }
  }
>
  <View>
    <View>
      <Text
        style={
          Object {
            "color": "#2E3192",
            "fontSize": 36,
            "marginBottom": 5,
          }
        }
      >
        Hi!
      </Text>
      <Text
        style={
          Object {
            "color": "#979797",
            "fontSize": 18,
          }
        }
      >
        Create a new account
      </Text>
    </View>
    <View
      style={
        Object {
          "height": "70%",
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Name"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Email Address"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Password"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Confirm password"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 30,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2E3192",
              "borderRadius": 5,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 18,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="register"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            REGISTER
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Given a Register component When some textInput is typing Then her onChange function will invoked 1`] = `
<RCTScrollView
  collapsable={false}
  keyboardShouldPersistTaps="handled"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "marginLeft": 50,
      "marginRight": 50,
      "paddingTop": 50,
    }
  }
>
  <View>
    <View>
      <Text
        style={
          Object {
            "color": "#2E3192",
            "fontSize": 36,
            "marginBottom": 5,
          }
        }
      >
        Hi!
      </Text>
      <Text
        style={
          Object {
            "color": "#979797",
            "fontSize": 18,
          }
        }
      >
        Create a new account
      </Text>
    </View>
    <View
      style={
        Object {
          "height": "70%",
          "width": "100%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Name"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value="HELLO"
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Email Address"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value="HELLO"
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Password"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value="HELLO"
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        onEndEditing={[Function]}
        placeholder="Confirm password"
        placeholderTextColor="#979797"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "borderBottomColor": "#979797",
            "borderBottomWidth": 1,
            "color": "#2E3192",
            "fontSize": 18,
            "height": 50,
            "marginBottom": 10,
            "marginTop": 10,
            "width": 300,
          }
        }
        underlineColorAndroid="transparent"
        value="HELLO"
      />
      <View
        style={
          Object {
            "height": 15,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 30,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2E3192",
              "borderRadius": 5,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 18,
              "opacity": 1,
              "width": "100%",
            }
          }
          testID="register"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
              }
            }
          >
            REGISTER
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
